### PR TITLE
Test against shapely pre-release

### LINKED ISSUE #678: test against `shapely==2.0b1`
We should test `spaghetti` against the `shapely` pre-release for future-proofing (add to CI matrix).

```
# using pip
pip install --pre shapely==2.0b1
# using conda /mamba
conda install -c conda-forge/label/shapely_dev shapely
```

xref pysal/spopt#289, @martinfleis

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR resolves  #678.

### CODE REVIEW

File: ci/310-DEV_shapely20b1.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+name: test
+channels:
+  - conda-forge
```
Comment (martinfleis) [inline ci/310-DEV_shapely20b1.yaml:3]:
```suggestion
  - conda-forge
  - conda-forge/label/shapely_dev
```

dev version is on a different channel

**Review by martinfleis — COMMENTED** [review]
I would probably just add one CI run with shapely 2.0. We want to catch behaviour changes and Python/OS versions should not make a huge difference.

Also, see the note on channels.

**Comment by martinfleis** [comment]
```
- package shapely-2.0b1-py37h649dd7e_0 is excluded by strict repo priority
```

It seems that micromamba is not happy about mixed channels. We can potentially install It from PyPI if it makes it easier. We do it in GeoPandas as well https://github.com/geopandas/geopandas/blob/403b7fa936bcae58134efbca51e10dce7a7c6639/ci/envs/39-latest-conda-forge.yaml#L39-L41

**Comment by martinfleis** [comment]
This actually seems to be more complicated than expected.

1. the env still shows 1.8.5 somehow
2. I've noticed that there is pygeos installed as well. In that case, we need to ensure that we have set and env variable `USE_PYGEOS=0` to ensure shapely is actually used. (https://geopandas.org/en/stable/getting_started/install.html#using-the-optional-pygeos-dependency)

**Comment by jGaboardi** [comment]
1. this is weird.
2. I can fix this.

**Comment by martinfleis** [comment]
Can we try the conda way again and set `channel-priority` to `flexible`? https://github.com/mamba-org/provision-with-micromamba#channel-priority

Hope it does not mess the other envs...

**Comment by jGaboardi** [comment]
[Got it](https://github.com/pysal/spaghetti/actions/runs/3340695955/jobs/5531090537#step:3:783) 🎉

File: ci/310-DEV_shapely20b1.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  - watermark
+  # testing/formatting
+  - black
+  - codecov
+  - pytest
+  - pytest-cov
+  - pytest-timeout
+  - pytest-xdist
+  # optional
+  - geopandas>=0.9.0
```
Comment (martinfleis) [inline ci/310-DEV_shapely20b1.yaml:22]:
```suggestion
  - geopandas>=0.12.0
```

Comment (martinfleis) [reply ci/310-DEV_shapely20b1.yaml:22]: This is a final bit. Then we can merge and potentially replicate the same PR elsewhere, where we don't use pygeos directly.

**Review by martinfleis — APPROVED** [review]

**Pull request merged**